### PR TITLE
[serde reflection] enable "tracing" of serde serializations

### PR DESCRIPTION
**Pull request by matbd**
## Motivation

This is a proof of concept to show how to extract IDL-like specifications from serde-based Rust code.

## Test Plan

cargo x test -p libra-serde-reflection
cargo run -p generate-format   # <-- a YAML description of Libra transactions

@boras-libra r=metajack

### CODE REVIEW

File: common/lcs/src/trace.rs [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+            return Err(merge_error(&*self, &other));
+        }
+        self.value.merge(other.value)
+    }
+}
+
+impl Merge for Format {
+    fn merge(&mut self, mut format: Format) -> Result<()> {
+        // Matching `&mut format` instead of `format` because of
+        // "error[E0009]: cannot bind by-move and by-ref in the same pattern"
```
Comment (metajack) [inline common/lcs/src/trace.rs]: You should include a link to https://github.com/rust-lang/rust/issues/68354 which tracks the stabilization of this feature, move_ref_pattern.

File: common/lcs/tests/serde.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 // For some reason deriving `Arbitrary` results in clippy firing a `unit_arg` violation
 #![allow(clippy::unit_arg)]
 
-use libra_canonical_serialization::{from_bytes, to_bytes, Error, MAX_SEQUENCE_LENGTH};
+use libra_canonical_serialization::{from_bytes, to_bytes, trace::*, Error, MAX_SEQUENCE_LENGTH};
```
Comment (metajack) [inline common/lcs/tests/serde.rs]: Should we  re-export the symbols in `trace::prelude` and `use trace::prelude::*` instead?

Comment (matbd) [reply common/lcs/tests/serde.rs]: Thanks. I removed the `::*` in a couple of places.

**Comment by matbd** [comment]
Thanks @metajack for the review.

I squashed the first diffs, applied your suggestions, fixed clippy errors, and rebased. So, this should be ready to land.

**Pull request closed without merging**